Weird bug with optionals, told me to report a bug

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.2 LTS (VM)
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
CC version: cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

getwd: /home/jordan/Desktop/Programming/polaris/core
vmodules: /home/jordan/.vmodules
vroot: /home/jordan/Desktop/Programming/Vtest/v
vexe: /home/jordan/Desktop/Programming/Vtest/v/v
vexe mtime: 2021-06-13 06:00:33
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 dc1c882.cab054c

Git version: git version 2.25.1
Git vroot status: weekly.2021.23-76-gcab054c1
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg message.v`
```v
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */

import crypto.sha1

// Topics are 64bit integers
struct Topic {
	hash u64
}

pub fn topic_from_name(topicName string) Topic {
	hashed := sha1.sum(topicName.bytes())
	mut num := u64(0)
	for i in 0 .. 8 {
		num |= hashed[i]<<(i<<3)
	}

	return Topic{
		hash: num
	}
}

// messages are simply anything which can be serialized
interface Message {
	t Topic
	serialize() ?[]byte
}

fn main() {
	println(topic_from_name("telem.loop.t1.j1.voltage"))
}


```

**What did you expect to see?**

no errors. the error goes away when I make serialize return a non-optional


**What did you see instead?**
```
==================
/tmp/v/message.9777363642470745560.tmp.c:1683: error: ';' expected (got "Option_Array_byte")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -1332,6 +1332,12 @@ fn (mut g Gen) stmt(node ast.Stmt) {
 		ast.Import {}
 		ast.InterfaceDecl {
 			// definitions are sorted and added in write_types
+			for method in node.methods {
+				if method.return_type.has_flag(.optional) {
+					// Register an optional if it's not registered yet
+					g.register_optional(method.return_type)
+				}
+			}
 		}
 		ast.Module {
 			// g.is_builtin_mod = node.name == 'builtin'

--- a/vlib/v/tests/interface_only_decl_with_optional_test.v
+++ b/vlib/v/tests/interface_only_decl_with_optional_test.v
@@ -1,0 +1,8 @@
+interface Message {
+	serialize() ?[]byte
+}
+
+fn test_interface_only_decl_with_optional() {
+	println('test interface')
+	assert true
+}

